federation-js/src/composition/validate/postComposition/__tests__/requiresFieldsMissingExternals.test.ts

```diff
@@ -68,6 +70,12 @@ describe('requiresFieldsMissingExternal', () => {
       Array [
         Object {
           "code": "REQUIRES_FIELDS_MISSING_EXTERNAL",
+          "locations": Array [
```

Comment:
This test might be a bit poorly constructed, but I see 2 possible cases here.

* In this case, `id` doesn't even exist in the extension. Not much we can do here. Maybe the location is the `fields` arg on the requires directive? I'm unsure.
* In what I would expect to be a more likely case, `id` would be defined but not marked as `@external`, in which case we could provide a location to that field instead since it's more relevant to the error.

Reply:
Since this is a post composition validator, there is no distinction between the fields on the Product of one service and the Product of another service, the `namedType.getFields()` returns the composed `Product`'s field (all of sku, upc, id and price). B/c of this, I am confused how we could find the node for the field on serviceB in this case at all. For now I made this the argument node.